### ISSUE TITLE
Conversion between `Value` and any `T: Serialize/Deserialize`

### ISSUE THREAD

**Issue by Diggsey** [ron-rs/ron#140]
Take a look at `serde_json::from_value` and `serde_json::to_value` - these functions would exactly mirror that.

RON has the unique opportunity to be an "impartial" serialization format for storing structured data internally in rust programs, so that it can still be converted to other formats without having to know the original type, but to do that, you need a way to go from arbitrary rust types to a RON value, and vice versa.

**Comment by torkleyy** [comment]
There already is a `Deserializer` implementation for `Value`, so this
shouldn't be very hard to implement.

**Comment by Boscop** [comment]
Yes, please add a `to_value` function! I need this for my use-case. (I'm dealing with large instances, and going through string is less efficient.)

(And `Value::into_rust` should be renamed to `from_value`.)

**Comment by torkleyy** [comment]
activity

**Comment by Kulasko** [comment]
Please please can we have this? I am using partial serialization/deserialization for automatic upgrading of file types and I really want support for a better format as JSON. The TOML crate doesn't support enums and tuples, so I arrived here. Having the functions like suggested by @Diggsey would make that usecane oh so much more painless.

**Comment by tiye** [comment]
can we add our own glue code to have `to_value` based on existing public functions? doesn't look like a core feature ron going to provide, but useful in some cases of data conversion.
